src/main/drivers/barometer/barometer_lps.c

```diff
@@ -0,0 +1,263 @@
+#define LPS_AVP_8               0x00
+#define LPS_AVP_32              0x01
+#define LPS_AVP_128             0x02
+#define LPS_AVP_512             0x03
+#define LPS_AVT_8               0x00
+#define LPS_AVT_16              0x04
+#define LPS_AVT_32              0x08
+#define LPS_AVT_64              0x0C
+//======================================Moving average number=============================================//
+#define LPS_NO_MA               0x00
+#define LPS_MA_2                0x01
+#define LPS_MA_4                0x03
+#define LPS_MA_8                0x07
+#define LPS_MA_16               0x0F
+#define LPS_MA_32               0x1F
+
+//Raw register values
+static uint32_t rawP = 0;
+static uint16_t rawT = 0;
+
+bool lpsWriteCommand(busDevice_t *busdev, uint8_t cmd, uint8_t byte) {
+    return spiBusWriteRegister(busdev, cmd, byte);
+}
+
+bool lpsReadCommand(busDevice_t *busdev, uint8_t cmd, uint8_t *data, uint8_t len) {
+    return spiBusReadRegisterBuffer(busdev, cmd | 0x80 | 0x40, data, len);
+}
+
+bool lpsWriteVerify(busDevice_t *busdev, uint8_t cmd, uint8_t byte) {
+    uint8_t temp = 0xff;
+    spiBusWriteRegister(busdev, cmd, byte);
+    spiBusReadRegisterBuffer(busdev, cmd, &temp, 1);
+    if (byte == temp) return true;
+    return false;
+}
+
+static void lpsOn(busDevice_t *busdev, uint8_t CTRL1_val) {
+    lpsWriteCommand(busdev, LPS_CTRL1, CTRL1_val | 0x80);
+    //Instead of delay let's ready status reg
+}
+
+static void lpsOff(busDevice_t *busdev) {
+    lpsWriteCommand(busdev, LPS_CTRL1, 0x00 | (0x01 << 2));
+}
+
+static void lps_nothing(baroDev_t *baro) {
+    UNUSED(baro);
+    return;
+}
+
+static void lps_read(baroDev_t *baro) {
+    uint8_t status = 0x00;
+    lpsReadCommand(&baro->busdev, LPS_STATUS, &status, 1);
+    if (status & 0x03) {
+        uint8_t temp[5];
+        lpsReadCommand(&baro->busdev, LPS_OUT_XL, temp, 5);
+
+        /* Build the raw data */
+        rawP = temp[0] | (temp[1] << 8) | (temp[2] << 16) | ((temp[2] & 0x80) ? 0xff000000 : 0);
+        rawT = (temp[4] << 8) | temp[3];
+    } else {
+        rawP = 0;
+        rawT = 0;
+    }
+}
+
+static void lps_calculate(int32_t *pressure, int32_t *temperature) {
+    *pressure = (int32_t)rawP * 100 / 4096;
+    *temperature = (int32_t)rawT * 10 / 48 + 4250;
+}
+
+bool lpsDetect(baroDev_t *baro)
+{
+    //Detect
+    busDevice_t *busdev = &baro->busdev;
+    IOInit(busdev->busdev_u.spi.csnPin, OWNER_BARO_CS, 0);
+    IOConfigGPIO(busdev->busdev_u.spi.csnPin, IOCFG_OUT_PP);
+    IOHi(busdev->busdev_u.spi.csnPin); // Disable
+    spiSetDivisor(busdev->busdev_u.spi.instance, SPI_CLOCK_STANDARD); // Baro can work only on up to 10Mhz SPI bus
+
+    uint8_t temp = 0x00;
+    lpsReadCommand(&baro->busdev, LPS_WHO_AM_I, &temp, 1);
+    if (temp != LPS25_ID && temp != LPS22_ID && temp != LPS33_ID && temp != LPS35_ID) {
+        return false;
+    }
+
+    //Init
+    lpsOff(busdev);
+    lpsWriteVerify(busdev, LPS_CTRL2, (0x00 << 1));
```

Comment:
`lpsWriteVerify` is called, but result is ignored .. is that intentional?

Reply:
Those registers from my experience always are written correctly, it's a bit redundant to check it, let's keep it as is anyway. Please

Reply:
`lpsWriteVerify` doesn't retry, it only returns `false` on mismatch. So no (visible) check is actually performed, unless debugger is used ... It is fine, but a bit strange